quattor/functions/strings.pan

```diff
@@ -0,0 +1,29 @@
+declaration template quattor/functions/strings;
+
+include { 'pan/functions' };
+
+#version comparison function (should not be here)
+function comparever = {
+  if(ARGC!=2){
+    error("comparever: wrong number of arguments. Required 2, given "+to_string(ARGC)+".");
+  };
+  version1=matches(ARGV[0],'(\d+)\.+(\d+)\.+(\d+)[-]*(\d*)');
```

Comment:
the pattern should be `^(\d+)\.(\d+)\.(\d+)(?:-(\d+))? ` or even `^(\d+)\.(\d+)(\.(\d+)(?:-(\d+))?)?`

and is best that you store the pattern in a variable so you are sure it's same pattern being matched.

but the remainder of the logic than also changes (a lot).

so maybe (untested)
```pan
pattern = '^(\d+)\.(\d+)(\.(\d+)(?:-(\d+))?)?';
version1 = matches(ARGV[0], pattern);
version2 = matches(ARGV[1], pattern);
len = length(version1);
if(len != length(version2)) {
    error(format("Compared versions are of different format (supported %s): %s vs %s", pattern, ARGV[0], ARGV[1]));
}
if (len == 0) {
    error(format("Compared versions do not match format %s: %s , %s", pattern, ARGV[0], ARGV[1]));
}
for (i=1; i <= len; i=i+1) {
...
} 
```